UI/ChatGPT/src/ChatGPT/Services/ChatService.cs

```diff
@@ -43,56 +64,47 @@ public async IAsyncEnumerable<ChatResponse> AskAsStream(ChatRequest chatRequest,
 		var response = new ChatResponse();
 		var content = new StringBuilder();
 
-		IAsyncEnumerator<ChatCompletionCreateResponse>? responseStream = default;
-		while (!response.IsError)
+		AsyncResultCollection<StreamingChatCompletionUpdate> result = _client.CompleteChatStreamingAsync(request);
+		var enumerator = result.GetAsyncEnumerator(ct);
+
+		try
 		{
-			try
+			while (await enumerator.MoveNextAsync() && !response.IsError)
 			{
-				responseStream ??= _client.CreateCompletionAsStream(request).GetAsyncEnumerator(ct);
-				if (await responseStream.MoveNextAsync())
+				var update = enumerator.Current;
+
+				try
 				{
-					if (responseStream.Current.Successful)
+					foreach (var updatePart in update.ContentUpdate)
 					{
-						foreach (var choice in responseStream.Current.Choices)
-						{
-							content.Append(choice.Message.Content);
-						}
-						response = response with { Message = content.ToString() };
-					}
-					else
-					{
-						response = response with { Message = responseStream.Current.Error?.Message, IsError = true };
+						content.Append(updatePart.Text);
 					}
+
+					response = response with { Message = content.ToString() };
 				}
-				else
+				catch (Exception ex)
 				{
-					yield break;
+					response = response with { Message = $"Something went wrong: {ex.Message}", IsError = true };
 				}
-			}
-			catch (Exception)
-			{
-				response = response with { IsError = true };
-			}
 
-			yield return response;
+				yield return response;
+			}
+		}
+		finally
+		{
+			await enumerator.DisposeAsync();
```

Comment:
This will no longer possible has you need to move it's creation back into the `try / catch` , but FYI instead of creating a `try / finally` for this, you could have use `***await using*** var enumerator = result.GetAsyncEnumerator(ct);`